charts/datafold/charts/nginx/templates/deployment.yaml

```diff
@@ -1,41 +1,73 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
+  name: {{ include "nginx.fullname" . }}
   labels:
-    app: nginx
-  name: nginx
+    {{- include "nginx.labels" . | nindent 4 }}
 spec:
-  progressDeadlineSeconds: 300
-  replicas: 1
-  revisionHistoryLimit: 3
   selector:
     matchLabels:
-      app: nginx
+      {{- include "nginx.selectorLabels" . | nindent 6 }}
+  # progressDeadlineSeconds: 300
+  replicas: {{ .Values.replicaCount }}
+  # revisionHistoryLimit: 3
   template:
     metadata:
-      creationTimestamp: null
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
       labels:
-        app: nginx
+        {{- include "nginx.labels" . | nindent 8 }}
+        {{- with .Values.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
     spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "nginx.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      terminationGracePeriodSeconds: {{ .Values.terminationGracePeriodSeconds }}
       containers:
-      - image: nginx
-        imagePullPolicy: Always
-        name: nginx
-        ports:
-        - name: nginx
-          containerPort: {{ .Values.global.nginx.port }}
-          protocol: TCP
-        resources:
-          limits:
```

Comment:
We have a story to deal with resource requests later. Let's keep everything clear. Also, we should not have added this hardcoded in the deployment but in the values yaml to keep everything consistent.